### ISSUE TITLE
MongoDB support

### ISSUE THREAD

**Issue by murilolobato** [creof/doctrine2-spatial#131]
Is MongoDB in plans to receive support? I'm trying to find something to help me to store GeoJSON points in mongo documents with symfony, but i haven't found anything yet.

Is this the proposal of this bundle?

**Comment by ddproxy** [comment]
As far as I can see, you should be able to store GeoJSON points in Mongo as json, the indexing properties are what would pose a problem.

@djlambert - Feel okay with adding MongoDB to our initiative?

**Comment by djlambert** [comment]
@ddproxy absolutely. After our previous discussion regarding GeoJSON this came to mind as an additional bonus.

**Comment by amenadiel** [comment]
That would be an epic achievement, boys. 

Keep in mindb that, while some datatypes and geometry primitives have their mongodb counterparts and while simple spatial predicates should be easily converted, is near unfeasible to operate with mixed quieres. 

Doing the intersections in memory is slow.

Perhaps is there a GEOS PHP module able to be offloaded the bigs guns

**Comment by ddproxy** [comment]
Intersections and other spatial operations should be done in memory (and definitely with indexed data) in the database.

https://github.com/phayes/geoPHP has the ability to use GEOS PHP for intersections. I would be concerned with the latency between fetching the data ( or big data ) from mongoDB and passing into the GEOS implementation of intersection. For larger operations the data redundancy would be huge - to the point that you'd want another layer between to locally cache the geospatial data from Mongo or even just save the shapes in shape-files with a pointer or summary in Mongo. But that's big-data and architecture concerns, if those proved to be issues, and if my worries are even close to being something of concern.

**Comment by amenadiel** [comment]
@ddproxy you're right, I actually can't recall having used doctrine2-spatial to find the intersections between two ArrayCollections that contain geometric entities.

So, the scope of supporting MongoDB would involve two aspects:
1. Adding custom functions to the [$DocumentManager](https://github.com/doctrine/mongodb-odm/blob/master/lib/Doctrine/ODM/MongoDB/DocumentManager.php) object as we currently do to the [$EntityManager](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/EntityManager.php) object.
2. Adapt our current custom datatypes to be usable with MongoDB.

I haven't seen MongoDB-ODM using DBAL, so it isn't obvious where are custom types added, if such a thing can be done.

**Comment by ddproxy** [comment]
@murilolobato 1.1.1 now has a toJson method that can be used to create the GeoJson representation for MongoDB. There are no ODM integration yet.
